src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -240,14 +243,32 @@ private void OnProjectChanged(object sender, IProjectContextInfo project)
                     return;
                 }
 
-                // get the list of projects
-                IEnumerable<IProjectContextInfo> projects = solutionModel.Projects.Select(p => p.NuGetProject);
-                Model.Context.Projects = projects;
+                var activityIsDisposed = false;
```

Comment:
A best practice for `IDisposable` is that it can be disposed of multiple times without any issues.

https://docs.microsoft.com/en-us/dotnet/api/system.idisposable.dispose?view=net-5.0#remarks

> If an object's Dispose method is called more than once, the object must ignore all calls after the first one. The object must not throw an exception if its Dispose method is called multiple times. Instance methods other than Dispose can throw an ObjectDisposedException when resources are already disposed.

I would get rid of this boolean that keeps track of whether or not the `activity` is disposed already and just do that work in the object itself.  Then you can safely dispose of it multiple times if needed.  I still think it would be better to have a `using` inside the `JTF.RunAsync()` if possible.